storage/tianmu/core/temp_table.h

```diff
@@ -319,7 +319,10 @@ class TempTable : public JustATable {
   uint GetDisplayableAttrIndex(uint attr);
   MultiIndex *GetMultiIndexP() { return filter.mind; }
   void ClearMultiIndexP() {
-    delete filter.mind;
+    if (!filter_shallow) {
```

Comment:
makes the `filter.mind = NULL` into `if` condition statment. otherwise, in-correcte use the function will lead memory leakage. it's not a defensive programming.